SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+from genericpath import isdir
+import sys
+import os
+import json
+import shutil
+from colorama import Fore
+from zipfile import ZipFile
+
+# Directory of all boards
+BOARDS_DIR = "boards/"
+# Other board files, which are same for all boards (there can be used placeholders)
+COMMON_DIR = "common/"
+# JSON file which describing the board (must be in each board directory)
+BOARD_GENFILE = "manifest.json"
+# Files, which are specific for each board. Will be copied from boards/BOARD/ directory
+BOARD_SPECIFIC_FILES = {"sdkconfig.defaults", "main/idf_component.yml"}
+# Generated output directory
+OUT_DIR = "espressif"
+# SquareLine output directory
+SQUARELINE_OUTDIR = "__ui_project_name__"
+# Template of SLB file
+SLB_FILE = {
+    "version":"",
+    "group":"Espressif",
+    "title": "",
+    "keywords": "Espressif",
+    "width": 0,
+    "height": 0,
+    "width_min": 0,
+    "height_min": 0,
+    "width_max": 0,
+    "height_max": 0,
+    "offset_x": 0,
+    "offset_y": 0,
+    "rotation": 0,
+    "color_depth": "",
+    "lvgl_export_path": "",
+    "lvgl_include_path": "lvgl.h",
+    "supported_lvgl_version": "8.0+",
+    "pattern_match_files": "./CMakeLists.txt",
+    "language":"C",
+    "ui_export_path":"./main/ui",
+    "url":"https://github.com/espressif/esp-bsp",
+    "short_description": "",
+    "long_description": ""
+}
+
+global placeholders
+
+# Print green text
+def print_ok(str):
+    print(Fore.GREEN + str + Fore.WHITE)
+
+# Print red error
+def print_error(str):
+    print(Fore.RED + str + Fore.WHITE)
+
+def remove_folder(path):
```

Comment:
You can use `shutil.rmtree` instead which does the same.

Reply:
The only usage seems to be `remove_folder(output + "/" + SQUARELINE_OUTDIR)`,  seems like it's okay to remove the entire SQUARELINE_OUTDIR directory? Why do we need to keep some subdirectories?

Reply:
Yes, I need only remove `out/board/SQUARELINE_OUTDIR` with everything inside. But when I do `shutil.rmtree(out/board/SQUARELINE_OUTDIR)`, it removes all, `SQUARELINE_OUTDIR` and then `board` and then `out` directory, which contains ZIP and other files.

Reply:
Huh, that's a pretty odd behavior. Let me check...